File: app/src/main/java/com/wild/corp/adhesion/models/Adherent.java
package com.wild.corp.adhesion.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Data
@EqualsAndHashCode(of = {"id"})
@Entity
@NoArgsConstructor
@Table(name = "adherents",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "id"),
        })
@JsonIgnoreProperties(ignoreUnknown = true,  allowSetters = true)
public class Adherent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String prenom;

    private String nom;

    private String genre;

    private String telephone;

    private LocalDate naissance;

    private String lieuNaissance;

    private String adresse;

    private String codePostal;

    private String ville;

    private Boolean adresseRepresentant;

    private Boolean telephoneRepresentant;

    private Boolean emailRepresentant;

    private Boolean mineur;

    private Boolean completAdhesion;

    @OneToMany(mappedBy = "adherent", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Accord> accords = new ArrayList<>();

    @ManyToOne
    @JsonIgnoreProperties({"cours", "accords", "adhesions", "activitesNm1", "tribu",  "documents", "derniereModifs", "derniereVisites", "representant"})
    private Adherent representant;

    @JsonIgnore
    @OneToMany(mappedBy = "adherentModif", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Notification> derniereModifs = new ArrayList<>();

    @OneToMany(mappedBy = "adherentVisite", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Notification> derniereVisites = new ArrayList<>();

    @ManyToOne
    @JsonIgnoreProperties({"adherents", "autorisations"})
    private Tribu tribu;

    @OneToMany(mappedBy = "adherent")
    @JsonIgnoreProperties({"paiements", "derniereModifs", "derniereVisites", "surClassement",  "adherent"})
    private Set<Adhesion> adhesions = new HashSet<>();

    @OneToMany(mappedBy = "adherent", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties({"adherent", "tribu"})
    private List<ActiviteNm1> activitesNm1 = new ArrayList<>();

    @ManyToMany
    @JsonIgnore
    private Set<Activite> cours = new HashSet<>();

    @OneToOne
    @JsonIgnoreProperties({"adherent", "tokens"})
    private User user;

}

File: client/src/app/models/adherent.ts
import { Accord } from "./accord";
import { Activite } from "./activite";
import { ActivitesNm1 } from "./activitesNm1";
import { Adhesion } from "./adhesion";
import { Document } from "./document";
import { Notification } from "./notification";
import { Tribu } from "./tribu";
import { User } from "./user";

export class Adherent {
  id!: number;
  prenom!: string;
  nom!: string;
  nomPrenom!: string;
  genre!: string;
  email!: string;
  telephone!: string;
  naissance!: Date;
  lieuNaissance!: string;
  adresse!: string;
  codePostal!: string;
  ville!: string;
  representant!: Adherent;
  adresseRepresentant!: boolean;
  telephoneRepresentant!: boolean;
  emailRepresentant!: boolean;

  mineur!: boolean;
  completAdhesion!: boolean;

  accords!: Accord[];
  documents: string[];
  derniereModifs!: Notification[];
  derniereVisites!: Notification[];
  tribu!: Tribu;
  adhesions!: Adhesion[]
  activitesNm1!: ActivitesNm1[]
  cours!: Activite[];
  user!: User;
  
  tribuId!: number
  editMail!: boolean
  rgpdDate!: Date;
  certifDate!: Date;
  edit!:boolean

  constructor() {

    this.cours = [];
    this.adhesions = [];
    this.accords = [];

    this.documents = []

    this.derniereVisites = []
    this.derniereModifs = []
  }



}
